### ISSUE TITLE
Mini BLE: check how udev rules are setup for Ubuntu

### ISSUE THREAD

**Issue by limpkin** [mooltipass/moolticute#506]
For the following 2 scenarios:
- fresh install
- update on current install, in the case only the mini udev rules are here and not the mini ble ones

**Comment by kaorihinata** [comment]
Good afternoon. This is Thomas from the Mooltipass BLE beta.

I spoke with the affected user on the mailing list earlier and I believe the issue he encountered was just a distribution incompatibility. At current, unless the user installs their own method of graphically elevating rights on Ubuntu, `data/moolticute.sh` simply falls back to sudo on AppImage execution, and in the absence of a terminal fails silently. The currently preferred method of handling this kind of escalation on Ubuntu (for applications that support it) is through GVFS, but for command line applications we should be able to make due with pkexec and env.

Something like the following should work (though I :

```
pkexec env DISPLAY="${DISPLAY}" XAUTHORITY="${XAUTHORITY}" sh << EOF
cp "$tmpfile" "${UDEV_RULES_FILE_PATH}" && \
    udevadm control --reload-rules
EOF
```

...or however you'd like to call it. That should produce a prompt on Ubuntu asking for admin credentials with a vague path describing the application in question. Not sure if calling env is still required (that [article I fished it out of](https://sites.google.com/site/installationubuntu/home/ubuntu-17-10/alternatives-for-gksu-and-gksudo) was 17.x era.) 

On a side note, unless there are plans to handle revisions later, replacing the grep for the check with a POSIX test + cmp should be enough and won't rely on human intervention to update the revision number.

Thanks,
Thomas

**Comment by limpkin** [comment]
@deXol task after CSV :)

**Comment by deXol** [comment]
Here (https://github.com/mooltipass/moolticute/commit/9dce6638a522932cf4c6bd304cfb32f22db6ee96) a check was added for whether the udev rule contains `"#version:2"` in v0.41.2-testing, however BLE related udev rule was only added later in v0.42.4-testing (https://github.com/mooltipass/moolticute/commit/3a2295776b1f60bbf57d2d18e3c9c76cb6b3630a)
So if the user started first or installed with v0.42.2-testing or v0.42.3-testing, then the new udev rule is not installed during startup.
Since these testing versions are not available since then, I do not think this can cause issue any more, but to make sure we can increase the version number to 3 in the udev rule.

**Comment by deXol** [comment]
### During installing
The script is checking if there is a sudo gui installed: https://github.com/mooltipass/moolticute/blob/master/data/moolticute.sh#L45
Based on that copying the udev rules: https://github.com/mooltipass/moolticute/blob/master/data/moolticute.sh#L73
If no gui is installed simply prompting for the sudo password and cp the file.

**Comment by kaorihinata** [comment]
I noticed the version check, which is why I added the note at the end:

> On a side note, unless there are plans to handle revisions later, replacing the grep for the udev rule check with a POSIX test + cmp should be enough and won't rely on human intervention to update the revision number.

The udev rules file in question is created solely for the purpose of allowing moolticute to access USB devices and is not intended to be customized by the user. What is actually in the file doesn't matter to moolticute as it doesn't parse it, so adding a version to it just increases the file size. I'm suggesting that if the file to install is different than the installed file you simply replace it as there doesn't appear to be any value (at the moment) to tracking the version of the installed udev rules file.
